apps/api/src/app/topics/e2e/filter-topics.e2e.ts

```diff
@@ -0,0 +1,103 @@
+import { UserSession } from '@novu/testing';
+import * as jwt from 'jsonwebtoken';
+import { expect } from 'chai';
+import { IJwtPayload, MemberRoleEnum } from '@novu/shared';
+
+const BASE_PATH = '/v1/topics';
+
+const createNewTopic = async (session: UserSession, topicKey: string) => {
+  return await session.testAgent.post(BASE_PATH).send({
+    key: topicKey,
+    name: `${topicKey}-name`,
+  });
+};
+
+describe('Filter topics - /topics (GET)', async () => {
+  let session: UserSession;
+
+  before(async () => {
+    session = new UserSession();
+    await session.initialize();
+
+    await createNewTopic(session, 'topic-key-1');
+    await createNewTopic(session, 'topic-key-2');
+    await createNewTopic(session, 'topic-key-3');
+  });
+
+  it('should return a validation error if the params provided are not in the right type', async () => {
+    const url = `${BASE_PATH}?page=first&pageSize=big`;
+    const response = await session.testAgent.get(url);
+
+    expect(response.statusCode).to.eql(400);
+    expect(response.body.error).to.eql('Bad Request');
+    expect(response.body.message).to.eql([
+      'page must be a number conforming to the specified constraints',
+      'pageSize must be a number conforming to the specified constraints',
+    ]);
+  });
+
+  it('should retrieve all the topics that exist in the database for the user if not query params provided', async () => {
```

Comment:
Wouldn't this be a heavy query in the future? Maybe we should force this endpoint to be always used with the pagination? wdyt?

Reply:
It is default limited to 100 entries, so the pagination is always there, and I do not find that will be so heavy (and if it is we have a huge problem inside the code 😓 ). I can lower the default in case you think it is too high.

Reply:
it's fine, haven't noticed the limit ;)